code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -101,11 +101,12 @@
 		return FALSE
 	if(T.density)
 		return TRUE
-	var/move_dir = 0
-	for(var/obj/structure/obstacle in T)
-		move_dir = get_dir(src, T)
-		if(obstacle.BlockedPassDirs(src, move_dir))
-			return TRUE
+	if(prob(20))
```

Comment:
what is the need of a prob 20 here

Reply:
Balance number, what is the chance that cades block the smoke

made into define